From 7ea930c5c41dcd076b521040fde63309f645a146 Mon Sep 17 00:00:00 2001
From: Yinhang Liu <yinhang.liu@intel.com>
Date: Fri, 21 Apr 2023 10:38:54 +0800
Subject: [PATCH 105/232] msdk: Add NumFilters check for VPP description

When NumFilters is equal to 0, the function should return
immediately.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4466>
---
 sys/msdk/gstmsdkcaps.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sys/msdk/gstmsdkcaps.c b/sys/msdk/gstmsdkcaps.c
index cd675a3cc..c6647e6e2 100644
--- a/sys/msdk/gstmsdkcaps.c
+++ b/sys/msdk/gstmsdkcaps.c
@@ -1365,8 +1365,11 @@ gst_msdkcaps_vpp_create_caps (GstMsdkContext * context,
   g_return_val_if_fail (context, FALSE);
   g_return_val_if_fail (vpp_description, FALSE);
 
-  session = gst_msdk_context_get_session (context);
   vpp_desc = (mfxVPPDescription *) vpp_description;
+  if (vpp_desc->NumFilters == 0)
+    return FALSE;
+
+  session = gst_msdk_context_get_session (context);
 
   g_value_init (&supported_in_fmts, GST_TYPE_LIST);
   g_value_init (&supported_out_fmts, GST_TYPE_LIST);
-- 
2.40.1

